### PR TITLE
OHRI-2050 replace momentjs with dayjs

### PR DESCRIPTION
**Pull request by pirupius**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary
- This PR removes momentjs and replaces it with dayjs for handling dates. 
- Updates framework dependencies

## Related Issue
https://ohri.atlassian.net/browse/OHRI-2050

### CODE REVIEW

**Review by ODORA0 — APPROVED** [review]

**Review by kajambiya — APPROVED** [review]
LGTM

File: packages/esm-commons-lib/package.json [hunk 1]
```diff
@@ -32,23 +32,21 @@
   },
   "dependencies": {
     "@carbon/react": "^1.13.0",
+    "@openmrs/openmrs-form-engine-lib": "next",
```
Comment (eudson) [inline packages/esm-commons-lib/package.json:35]: @pirupius why the `esm-commons-lib` have to depend on `the form-engine-lib`

Comment (eudson) [reply packages/esm-commons-lib/package.json:35]: I see you removed from the devDependencies but I still don't understand.

File: packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx [hunk 2]
```diff
@@ -189,7 +189,7 @@ const CurrentPregnancy: React.FC<PatientChartProps> = ({ patientUuid }) => {
     });
 
     return items;
-  }, [relatives, relativeToIdentifierMap, infantOutcomes]);
+  }, [relatives, patientUuid, relativeToIdentifierMap, infantOutcomes]);
```
Comment (eudson) [inline packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx:192]: why are you adding `patientUuid` here?

File: packages/esm-commons-lib/package.json (hunk 1 above)
Comment (pirupius) [reply packages/esm-commons-lib/package.json:35]: The encounter list component opens the form with the form-engine-lib

Comment (pirupius) [reply packages/esm-commons-lib/package.json:35]: About us removing it from devDependencies. We agreed during Friday scrum to make it a dependency to fix issues where the versions of the form engine were not aligned.

File: packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx (hunk 2 above)
Comment (pirupius) [reply packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx:192]: There's caching of this value so we want to force a refresh if the patientUuid changes

**Pull request merged**